src/Raven.Server/Documents/DatabaseRequestHandler.cs

```diff
@@ -242,6 +247,42 @@ private DynamicJsonValue GetConnectionStringConfigurationAuditJson(BlittableJson
             return null;
         }
 
+        protected void IsAllowedToExecuteExternalConfigurationScript(string name, BlittableJsonReaderObject configuration)
+        {
+            X509Certificate2 clientCert = GetCurrentCertificate();
+            CertificateDefinition newCertificateDef;
+
+            if (clientCert == null)
+                return;
+
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
+            using (ctx.OpenReadTransaction())
+            {
+                BlittableJsonReaderObject certificate = ServerStore.Cluster.GetCertificateByThumbprint(ctx, clientCert.Thumbprint) ??
+                                                        ServerStore.Cluster.GetLocalStateByThumbprint(ctx, clientCert.Thumbprint);
+                newCertificateDef = JsonDeserializationServer.CertificateDefinition(certificate);
+            }
+
+            List<BackupSettings> settingsList = null;
+            if (name == OngoingTasksHandler.PutConnectionStringDebugTag && ConnectionString.GetConnectionStringType(configuration) == ConnectionStringType.Olap)
+                settingsList = JsonDeserializationClient.OlapConnectionString(configuration).GetBackupSettingsDestinations();
+
+            if (name is OngoingTasksHandler.UpdatePeriodicBackupDebugTag or OngoingTasksHandler.BackupDatabaseOnceTag)
+                settingsList = JsonDeserializationServer.BackupConfiguration(configuration).GetBackupSettingsDestinations();
+
+            if (settingsList == null) return;
+            foreach (var dest in settingsList)
+            {
+                if (dest.GetBackupConfigurationScript != null)
```

Comment:
can `dest` be null?

Reply:
checked again ,  and `dest` can't be null. the check being made earlier in `GetBackupSettingsDestinations` method. if some destination will be null ,it won't make it to the list.